pkg/runtime/pubsub/subscriptions.go

```diff
@@ -0,0 +1,74 @@
+package pubsub
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+
+	"github.com/dapr/dapr/pkg/channel"
+	"github.com/dapr/dapr/pkg/logger"
+	invokev1 "github.com/dapr/dapr/pkg/messaging/v1"
+	daprclientv1pb "github.com/dapr/dapr/pkg/proto/daprclient/v1"
+	"github.com/golang/protobuf/ptypes/empty"
+)
+
+const (
+	getTopicsError         = "error getting topic list from app: %s"
+	deserializeTopicsError = "error getting topics from app: %s"
+	noSubscriptionsError   = "user app did not subscribe to any topic"
+)
+
+func GetSubscriptionsHTTP(channel channel.AppChannel, log logger.Logger) []Subscription {
+	var subscriptions []Subscription
+	req := invokev1.NewInvokeMethodRequest("dapr/subscribe")
+	req.WithHTTPExtension(http.MethodGet, "")
+	req.WithRawData(nil, invokev1.JSONContentType)
+
+	// TODO Propagate Context
+	ctx := context.Background()
+	resp, err := channel.InvokeMethod(ctx, req)
+	if err != nil {
+		log.Errorf(getTopicsError, err)
+	}
+
+	switch resp.Status().Code {
+	case http.StatusOK:
+		_, body := resp.RawData()
+		if err := json.Unmarshal(body, &subscriptions); err != nil {
+			log.Errorf(deserializeTopicsError, err)
+		}
+	case http.StatusNotFound:
+		log.Debug(noSubscriptionsError)
+
+	default:
+		log.Warnf("app returned http status code %v from subscription endpoint", resp.Status().Code)
+	}
+	for i := len(subscriptions) - 1; i >= 0; i-- {
+		if subscriptions[i].Route == "" {
+			subscriptions = append(subscriptions[:i], subscriptions[i+1:]...)
+			log.Warnf("topic %s has an empty route. removing from subscriptions list", subscriptions[i].Topic)
```

Comment:
Thanks for adding a check for it.
subscriptions size is being reduced in the loop, loop is going till the original length of subscriptions.
Also subscriptions[i].Topic will return incorrect topic name as subscriptions has been modified.
Please add a unit test for it.

Reply:
This is the way in Go to modify a loop while iterating. The log warning needs to go before the append though.

Reply:
@yaron2 can you double check this logic? If route is empty at len(subs)-1 index, it would  panic because subs[i+1:] points out of index.

Reply:
> @yaron2 can you double check this logic? If route is empty at len(subscriptions)-1 index, it will panic with out of index error because subscriptions[i+1:] points len(subscriptions) index

oh nevermind. no error...

Reply:
Interesting... my test code was panic because of this :)